### ISSUE TITLE
parsing failed when using `parserOptions.project` with tsx

### ISSUE THREAD

**Issue by 101polarbears** [vuejs/eslint-plugin-vue#2479]
**Checklist**

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have read the [FAQ](https://eslint.vuejs.org/user-guide/#faq) and my problem is not listed.

**Tell us about your environment**

- **ESLint version: 8.57.0**
- **eslint-plugin-vue version: 9.22.0**
- **Vue version: 3.4.21**
- **Node version: 18.14.1**
- **Operating System: Windows 11**

**Please show your full configuration:**

```js
/** @type {import("eslint").ESLint.ConfigData} */
module.exports = {
  root: true,
  overrides: [
    {
      files: ["*.vue"],
      plugins: ["@typescript-eslint"],
      parser: "vue-eslint-parser",
      parserOptions: {
        parser: {
          tsx: "@typescript-eslint/parser",
          "<template>": "espree"
        },
        extraFileExtensions: [".vue"],
        project: ['./tsconfig.eslint.json'],
        ecmaFeatures: {
          jsx: true,
        },
      },
    },
  ],
};
```

**What did you do?**	

```vue
<template>
  <div></div>
</template>

<script lang="tsx" setup>
const VARRIABLE_NAME = '';
const x = <span>{VARRIABLE_NAME}</span>;
</script>
```

**What did you expect to happen?**
Eslint should output nothing.

**What actually happened?**

![error_img](https://github.com/vuejs/eslint-plugin-vue/assets/23090503/97c9082b-2bf7-4e67-9a46-df1122ef1c22)

**Repository to reproduce this issue**

https://stackblitz.com/edit/vitejs-vite-4napuv?file=index.vue&view=editor

The eslint output looks exactly like `parserOptions.ecmaVersions.jsx = false`
*removing the `parserOptions.project` config fix this issue, but losses the ability to get type informed eslint checks*
Other dependencies:
- `@typescript-eslint/eslint-plugin`@6.21.0/7.13.1
- `@typescript-eslint/parser`@6.21.0/7.13.1

here's the `tsconfig.eslint.json` referenced in `parserOptions.project`
```json
{
  "compilerOptions": {
    "jsx": "preserve",
    "jsxImportSource": "vue",
    "module": "ESNext",
    "noEmit": true,
    "skipLibCheck": true,
    "target": "ESNext"
  },
  "include": ["./index.vue"]
}
```

**Comment by FloEdelmann** [comment]
See <https://typescript-eslint.io/troubleshooting/#i-am-running-into-errors-when-parsing-typescript-in-my-vue-files> and <https://eslint.vuejs.org/user-guide/#what-is-the-use-the-latest-vue-eslint-parser-error>.

**Comment by 101polarbears** [comment]
> See https://typescript-eslint.io/troubleshooting/#i-am-running-into-errors-when-parsing-typescript-in-my-vue-files and https://eslint.vuejs.org/user-guide/#what-is-the-use-the-latest-vue-eslint-parser-error.

Already did that. I didn't see anything referring to using these two things together. I just put together all the configs I found, and get the error.

**Comment by FloEdelmann** [comment]
Oh, you're right. But nevertheless, this seems to be an issue on typescript-eslint's side rather than eslint-plugin-vue's side. See also (emphasis theirs):

> For "unknown" extensions (`.md`, `.vue`):
>
> * If `parserOptions.project` is not provided:
>   * The setting will be respected.
> * If `parserOptions.project` is provided (i.e. you are using rules with type information):
>   * **always parsed as if this is `false`**

https://typescript-eslint.io/packages/parser/#jsx

**Comment by 101polarbears** [comment]
Fair enough, thanks for your help
